Tasks/AzureRmWebAppDeployment/Utility.ps1

```diff
@@ -216,4 +218,100 @@ function Get-UserAgentString
 	}
 	
 	return $userAgent
+}
+
+function Update-DeploymentStatus
+{
+
+    param([Parameter(Mandatory=$true)] $azureRMWebAppConnectionDetails,
+          [Parameter(Mandatory=$true)] $deployAzureWebsiteError)
+
+        $webAppPublishKuduUrl = $azureRMWebAppConnectionDetails.KuduHostName
+        if ($webAppPublishKuduUrl) {
+            $status = 3 #failed
+            $status_text = "failed"
+            if(!$deployAzureWebsiteError) {
+                $status = 4 #succeeded
+                $status_text = "succeeded"
+            }
+
+            $username = $azureRMWebAppConnectionDetails.UserName
+            $securePwd = ConvertTo-SecureString $azureRMWebAppConnectionDetails.UserPassword -AsPlainText -Force
+            $credential = New-Object System.Management.Automation.PSCredential ($username, $securePwd)
+
+            $author = Get-VstsTaskVariable -Name "build.sourceVersionAuthor"
+            if([string]::IsNullOrEmpty($author)) {
+                # fall back to build/release requestedfor
+                $author = Get-VstsTaskVariable -Name "build.requestedfor"
+                if([string]::IsNullOrEmpty($author)) {
+                    $author = Get-VstsTaskVariable -Name "release.requestedfor"
+                }
+                # At this point if this is still null, let's use agent name
+                if([string]::IsNullOrEmpty($author)) {
+                    $author = Get-VstsTaskVariable -Name "agent.name"
+                }
+            }
+
+            # using buildId/releaseId to update deployment status
+            # using buildUrl/releaseUrl to update deployment message
+            $buildUrlTaskVar = Get-VstsTaskVariable -Name "build.buildUri"
+            $releaseUrlTaskVar = Get-VstsTaskVariable -Name "release.releaseUri"
+            $buildIdTaskVar = Get-VstsTaskVariable -Name "build.buildId"
+            $releaseIdTaskVar = Get-VstsTaskVariable -Name "release.releaseId"
+            if(-not [string]::IsNullOrEmpty($releaseUrlTaskVar)) {
+                $deploymentId = $releaseIdTaskVar
+                $message = Get-VstsLocString -Key "Updatingdeploymenthistoryfordeployment0" -ArgumentList $releaseUrlTaskVar
+            }
+            else
+            {
+               $deploymentId = $buildIdTaskVar
+               $message = Get-VstsLocString -Key "Updatingdeploymenthistoryfordeployment0" -ArgumentList $buildUrlTaskVar
+            }
+
+            Write-Verbose "Using deploymentId as: '$deploymentId' to update deployment Status"
+            Write-Verbose "Using message as: '$message' to update deployment Status"
+
+            if([string]::IsNullOrEmpty($deploymentId)) {
+                #No point in proceeding further
+                Write-Warning (Get-VstsLocString -Key "Cannot update deployment status, unique deploymentId cannot be retrieved")  
+                Return
+            }
+
+            $collectionUrl = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI".TrimEnd('/')
```

Comment:
Use $collectionUrl = Get-VstsTaskVariable -Name System.TeamFoundationCollectionUri -Require 
and similar for rest